### PR TITLE
Backup SCCs using SA plugin

### PR DESCRIPTION
**Pull request by alexander-demicev**
This PR adds backup plugin for service accounts. It's purpose is to backup SCCs as additional items for service accounts. SCCs are cluster scoped resources and this means that we need to establish a relation with service accounts, this could be done in the same way as in CPMA(https://github.com/fusor/cpma/blob/master/pkg/transform/cluster/cluster.go#L520), by looking at SCC name.

### CODE REVIEW

**Comment by sseago** [comment]
So there's a problem with 'dep ensure' pulling in symlinks under vendor/ that we don't want. This causes problems when trying to use this on non-linux checkouts. Basically every time you run `dep ensure` you need to run `remove-known-symlinks-after-dep-ensure.sh` which removes a specific set of symlinks we expect to show up there. Then you can run `find-vendor-symlinks.sh` to make sure there are none left. These shouldn't be added to the PR/build.

File: velero-plugins/migsa/backup.go
Comment (sseago) [inline velero-plugins/migsa/backup.go:92]: Do we need to take splitUsername[2] into account here, which seems to be the namespace where the SA is relevant? Since SAs are namespaced, should we only copy SCCs where serviceaccount name (splitUsername[3]) *and* namespce (splitUsername[2]) match?

File: velero-plugins/main.go
Comment (sseago) [inline velero-plugins/main.go]: This should be `RegisterBackupItemAction`

File: velero-plugins/migsa/backup.go
Comment (alexander-demicev) [reply velero-plugins/migsa/backup.go:92]: You are right, I will fix it. Thanks!

File: velero-plugins/migsa/backup.go
Comment (sseago) [inline velero-plugins/migsa/backup.go:60]: If you try to migrate a namespace for which there are no SCCs defined, will `p.SCCMap[serviceAccount.Namespace]` be nil, and if so, will this result in a segfault? We should probably make sure that the map entry for the namespace exists before trying to look up the serviceaccount name in it.

**Review by sseago — COMMENTED** [review]
I think this looks reasonable overall, except for the need for nil checking around the SCC map lookup.

File: velero-plugins/migsa/backup.go
Comment (dymurray) [inline velero-plugins/migsa/backup.go]: Does this check need to be `if len(splitUsername) <= 2`? Since below we are grabbing elements 2 and 3 from the array.

File: velero-plugins/migsa/backup.go
Comment (dymurray) [inline velero-plugins/migsa/backup.go:82]: I mentioned privately to Alex that this PR will not solve the use case of migrating SCCs for service accounts that are included as part of the `scc.Groups` field. We will submit a follow-up PR to solve this.

File: velero-plugins/main.go
Comment (dymurray) [inline velero-plugins/main.go]: My only concern with this is that we are only establishing the SCC map when the velero plugin is registered. If any SCC changes are made after the velero pod starts up then we will be working with stale data. Perhaps in the plugin itself we have an `UpdateSCCMap()` call that refreshes the data?

I don't consider this a blocker for merging this PR because I don't see this being a huge problem. SCCs aren't touched very often.

**Review by dymurray — APPROVED** [review]
I'm going to approve this but had a couple comments. Since we are going to have a follow up PR for groups I'm fine with solving the `UpdateSCCMap` comment there, but I would prefer if you update the safety check on the array before we merge.

Great job on this!

File: velero-plugins/main.go
Comment (sseago) [reply velero-plugins/main.go]: Is there an easy way to make this update for each backup without having to regenerate it for each service account?

File: velero-plugins/migsa/backup.go
Comment (sseago) [reply velero-plugins/migsa/backup.go]: Good catch. Tthe first version of this PR only looked at the first two elements, so this should be changed now.

**Review by sseago — APPROVED** [review]
Approved, assuming the `if len(splitUsername)` fix is made, as suggested below. Also, if we keep the SCC map initialization only at plugin registration, we need to document somewhere that velero has to be bounced after new SCCs are created. Alternatively, it might be easier to just find a better approach to initializing this map.

File: velero-plugins/main.go
Comment (alexander-demicev) [reply velero-plugins/main.go]: I can try to set up a watcher and update SCCMap when needed

**Review by dymurray — APPROVED** [review]
ACKing update SCC map changes

**Review by sseago — APPROVED** [review]
LGTM

**Pull request merged**
